src/sentry/replays/usecases/reader.py

```diff
@@ -0,0 +1,256 @@
+from __future__ import annotations
+
+import uuid
+import zlib
+from concurrent.futures import ThreadPoolExecutor
+from datetime import datetime, timedelta
+from typing import List
+
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Granularity,
+    Limit,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry.replays.lib.storage import FilestoreBlob, RecordingSegmentStorageMeta, StorageBlob
+from sentry.replays.models import ReplayRecordingSegment
+from sentry.utils.snuba import raw_snql_query
+
+# METADATA QUERY BEHAVIOR.
+
+
+def fetch_segments_metadata(
+    project_id: int,
+    replay_id: str,
+    offset: int,
+    limit: int,
+) -> List[RecordingSegmentStorageMeta]:
+    """Return a list of recording segment storage metadata."""
+    # NOTE: This method can miss segments that were split during the deploy.  E.g. half were on
```

Comment:
i wonder if we could modify ingestion a little bit to take care of this. something like "if org in set to try new storage, check to see that current replay  (segment_id  > 0) doesn't have any previous segments stored in the old system. and if it doesn't, go ahead and write to direct storage. 

there are a few different possibilities to cache this value as well

Reply:
Experiment with Postgres first.  If it has problems we can try Postgres.

Reply:
This will be on the writer PR.